src/pages/task/TaskTabs.tsx

```diff
@@ -0,0 +1,206 @@
+
+    [TaskTab.Tests]: (
+      <Tab
+        name={
+          <span>
+            {failedTestCount ? (
+              <TabLabelWithBadge
+                tabLabel="Tests"
+                badgeVariant="red"
+                badgeText={failedTestCount}
+                dataCyBadge="tests-tab-badge"
+              />
+            ) : (
+              "Tests"
+            )}
+          </span>
+        }
+        data-cy="task-tests-tab"
+        key="task-tests-tab"
+      >
+        <TestsTable />
+      </Tab>
+    ),
+    [TaskTab.ExecutionTasks]: (
+      <Tab
+        name="Execution Tasks"
+        data-cy="task-execution-tab"
+        key="execution-tasks-tab"
+      >
+        <TasksTable
+          tasks={executionTasksFull}
+          onClickTaskLink={() =>
+            taskAnalytics.sendEvent({ name: "Click Execution Task Link" })
+          }
+        />
+      </Tab>
+    ),
+    [TaskTab.Files]: (
+      <Tab
+        name={
+          <span>
+            {fileCount !== undefined ? (
+              <TabLabelWithBadge
+                tabLabel="Files"
+                badgeVariant="lightgray"
+                badgeText={fileCount}
+                dataCyBadge="files-tab-badge"
+              />
+            ) : (
+              "Files"
+            )}
+          </span>
+        }
+        data-cy="task-files-tab"
+        key="task-files-tab"
+      >
+        <FilesTables />
+      </Tab>
+    ),
+    [TaskTab.Annotations]: (
+      <Tab
+        name="Task Annotations"
+        data-cy="task-build-baron-tab"
+        key="task-build-baron-tab"
+      >
+        <BuildBaron
+          annotation={annotation}
+          bbData={buildBaronData}
+          error={buildBaronError}
+          taskId={id}
+          execution={execution}
+          loading={buildBaronLoading}
+          userCanModify={canModifyAnnotation}
+        />
+      </Tab>
+    ),
+    [TaskTab.TrendCharts]: (
+      <Tab
+        name="Trend Charts"
+        data-cy="trend-charts-tab"
+        key="trend-charts-tab"
+      >
+        <TrendChartsPlugin taskId={id} />
+      </Tab>
+    ),
+  };
+
+  const tabIsActive = {
+    [TaskTab.Logs]: !isDisplayTask,
+    [TaskTab.ExecutionTasks]: isDisplayTask,
+    [TaskTab.Tests]: true,
+    [TaskTab.Files]: true,
+    [TaskTab.Annotations]: showAnnotationsTab,
+    [TaskTab.TrendCharts]: isPerfPluginEnabled,
+  };
+
+  const activeTabs = Object.keys(tabMap).filter((tab) => tabIsActive[tab]);
+
+  let defaultTab = 0;
+  if (urlTab && activeTabs.indexOf(urlTab) > -1) {
+    defaultTab = activeTabs.indexOf(urlTab);
+  } else if (isDisplayTask) {
+    defaultTab = activeTabs.indexOf(TaskTab.ExecutionTasks);
+  } else if (totalTestCount > 0) {
+    defaultTab = activeTabs.indexOf(TaskTab.Tests);
+  }
+  const [selectedTab, setSelectedTab] = useState(defaultTab);
+
+  // This is used to keep track of the first tab transition so we dont accidently trigger an analytics event for it
+  const firstRender = useRef(true);
```

Comment:
another option is to track the previous `id` with `usePrevious` and only track analytics is prev is undefined and curr isn't